main.sh

```diff
@@ -184,25 +222,16 @@ while true; do
         fi
 
         echo "uploading: ${dir}"
-        # ensure upload file parent dirs exists
-        ssh beehive-upload-server "mkdir -p ~/uploads/${dir}/"
-        rsync -av \
-            --exclude '.tmp*' \
-            --progress \
-            --compress \
-            --remove-source-files \
-            --itemize-changes \
-            --partial-dir=.partial/ \
-            --bwlimit=0 \
-            "${dir}/" \
-            "beehive-upload-server:~/uploads/${dir}/"
+        upload_dir "${dir}"
```

Comment:
So this seems to behave the same as it did before this change _EXCECPT_ for
1) on error code 20, we retry?  can this get stuck forever?  do we care?
2) on any other error code, we return 1 here

what are we doing with the error code that is returned from `upload_dir`?  Do we need to do anything to handle it in this outer loop?  i didn't know if this bash script was set to exit on error or not.